react/lifecycle/trigger-action-on-props-update.mo.md

```diff
@@ -1,110 +1,252 @@
-# [MO] Trigger action on props update with componentWillReceiveProps
+# [MO] Trigger action on props update with componentDidUpdate and getDerivedStateFromProps
 
-## Owner: Yassine Chbani
+## Owner: Nicolas Djambazian
 
 ## Motivation
 
 If you have ever worked with React, you will have probably used Redux at some point and found it great as your components can
 subscribe to any changes to the store state and react accordingly. But what happens if you want to similarly handle the update
 of one of your component's props? For example, call a certain method for a given change of the props?
 
-Well you're in luck, because Facebook provides a set of methods that get called at different points of the lifecycle of a component <sup>1</sup>. We are going to look at the `ComponentWillReceiveProps`method in the following article.
+Your component has a set of props that can change after it has been mounted and before it gets unmounted. The props change can be
+triggered from within or outside the component.
 
+Well you're in luck, because Facebook provides a set of methods that get called at different points of the lifecycle of a component <sup>1</sup>. We are going to look at the `getDerivedStateFromProps` and  the `componentDidUpdate` methods in the following article. The first one is useful if you want to change your state according to the props. The second, if you want to trigger actions on props or state change.
 
-## Prerequisites
 
-Your page has a set of props that can change after it has been mounted and before it gets unmounted. The props change can be
-triggered from within or outside the page.
+## Updating the state from a props (~10 minutes)
 
-## Steps (~10 minutes)
+### Prerequisites
+
+For the example, let's say we have an input text which saves its value in its internal state, but we want to allow the parent to control him sending a `value` props.
 
-- Add to your prop a component that gets updated from outside the page e.g. from a Redux store:
 
 ```jsx
-import { Connectivity } from '../../../Connectivity';
+type Props = { };
+type State = { value: string };
+class Input extends Component<Props, State> {
+  state = {
+    value: '',
+  };
+
+  onChange = (value: string) => {
+    this.setState({ value });
+  }
+
+  render () {
+    return (
+      <SomeInputComponent value={this.state.value} onChange={this.onChange} />
+    )
+  }
+}
+```
+
+### Steps
+
+In the previous version of React, we used to implement the `componentWillReceiveProps` for that use case. Since react rendering will be asynchronous, this method is deprecated. You now need to use the static method `getDerivedStateFromProps`.
+
```

Comment:
code smell : https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html